client/post-editor/media-modal/index.jsx

```diff
@@ -111,7 +111,13 @@ export class EditorMediaModal extends Component {
 	isDisabled() {
 		return some( this.props.mediaLibrarySelectedItems, function( item ) {
 			const mimePrefix = MediaUtils.getMimePrefix( item );
-			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.props.view );
+			return item.transient && (
+				// Transients can't be handled by the editor if they are being
+				// uploaded via an external URL
+				item.is_uploading_via_url ||
```

Comment:
> The downside I see to this approach is that it may mislead a developer to think this is a real property of the media endpoints response payload.

True. Though, for me:

> If we know whether we'd be able to insert anything by checking `media.URL`

this is relying on the implementation of the layer below, in a sense. I'm torn between both approaches, but if I had to choose now I'd keep the current one.

> mislead a developer to think this is a real property

What if we prefix the property with `calypso_`, would that mitigate it?